lib/math.ex

```diff
@@ -472,4 +472,65 @@ defmodule Math do
   """
   @spec atanh(x) :: float
   defdelegate atanh(x), to: :math
+
+  @doc """
+  Computes the modular multiplicatibe inverse of `a` under modulo `m`
+
+  In other words, given integers `a` and `m` calculate a value `b` for `ab = 1 (mod m)`
+
+  Returns an `{:ok, b}` tuple or `{:error, :not_coprime}` when `b` cannot be calculated for the inputs.
+
+  ## Examples
+
+      iex> Math.mod_inv 3, 11
+      {:ok, 4}
+      iex> Math.mod_inv 10, 17
+      {:ok, 12}
+      iex> Math.mod_inv 123, 455
+      {:ok, 37}
+      iex> Math.mod_inv 123, 456
+      {:error, :not_coprime}
+
+  """
+  @spec mod_inv(a :: integer, m :: integer) :: {:ok, integer} | {:error, :not_coprime}
+  def mod_inv(a, m)
+
+  def mod_inv(a, m) do
```

Comment:
To create a more clear error message when someone mistakenly tries to call this function with non-integers (like a binary or a map), I think it would be a good idea to add a `when is_integer(a) and is_integer(m)` guard clause to this.

Otherwise the error will only happen on line 499 or 500 and be much less clear.

Reply:
While a bit ugly it does make sense.